### PR TITLE
Adding a method to select the specific area to be framed

### PR DESCRIPTION
**Pull request by maxfish**

### CODE REVIEW

File: pkg/graphics/camera_2d.go [hunk 1]
```diff
@@ -66,6 +67,19 @@ func (c *Camera2D) SetFlipVertical(flip bool) {
...
 }
 
+// SetVisibleArea configures the camera to make the specified area completely visible, position and zoom are changed accordingly
+func (c *Camera2D) SetVisibleArea(x1 float32, y1 float32, x2 float32, y2 float32) {
+	width := float32(math.Abs(float64(x2 - x1)))
+	height := float32(math.Abs(float64(y2 - y1)))
+	zoom := float32(math.Min(float64(c.width/width), float64(c.height/height)))
+	c.SetZoom(zoom)
+	if c.centered {
+		c.SetPosition(x1+width/2, y1+height/2)
```
Comment (keymone) [inline pkg/graphics/camera_2d.go]: we should either assert that `x1<x2` or `SetPosition` based on `x2` if it's smaller

**Comment by maxfish** [comment]
I've also added a test to cover the edge case
